plugins/pipelines/app/models/concerns/pipeline_plugin_concern.rb

```diff
@@ -0,0 +1,33 @@
+module PipelinePluginConcern
+  # Return true if any stages in the pipeline are marked production
+  def production?
+    super || Stage.find(next_stage_ids).any? { |s| s.production? }
```

Comment:
Yup, to the .any?(&:production?), but it also seems activerecord is smart enough to skip the find if passed an empty array.